src/Messenger/KafkaTransport.php

```diff
@@ -135,7 +152,28 @@ public function ack(Envelope $envelope): void
 
     public function reject(Envelope $envelope): void
     {
-        // Do nothing. auto commit should be set to false!
+        /** @var KafkaMessageStamp $transportStamp */
+        $transportStamp = $envelope->last(KafkaMessageStamp::class);
+        $message = $transportStamp->getMessage();
+
+        $this->rejectKafkaMessage($message);
+    }
+
+    private function rejectKafkaMessage(Message $message): void
+    {
+        if (false === $this->ackFailedMessages) {
+            return;
+        }
+
+        $consumer = $this->getConsumer();
```

Comment:
@KonstantinCodes this depends on the question below, if there is always an envelope with correct serialized class it can use the ack() if you think its a good idea to catch the serialization exception and ack the message + continue then its not possible.